sqlair.go

```diff
@@ -92,15 +97,98 @@ type Iterator struct {
 	started bool
 }
 
+// A Least Recently Used Cache using a doubly linked list.
+type lruCache struct {
+	ll   *list.List
+	c    map[index]*list.Element
+	size int
+}
+
+// The cache key.
+type index struct {
+	tx *TX
+	s  *Statement
+}
+
+// The cache value.
+type entry struct {
+	key   index
+	value *sql.Stmt
+}
+
+func newLRUCache(size int) *lruCache {
+	c := &lruCache{}
+	c.ll = list.New()
+	c.c = make(map[index]*list.Element)
+	c.size = size
+	return c
+}
+
+func (c *lruCache) lookup(s *Statement, tx *TX) (*sql.Stmt, bool) {
+	e, ok := c.c[index{tx: tx, s: s}]
+	if !ok {
+		return nil, ok
```

Comment:
I think this return would be slightly clearer as `return nil, false` as there's only one value `ok` could be here

Reply:
Good point.